### ISSUE TITLE
Wildcard search for CJK characters with ? does not match exactly one character unless there are multiple ?

### ISSUE THREAD

**Issue by lrorpilla** [isar/isar#1135]
### Steps to Reproduce

1. Use `matches` with CJK characters.
2. Observe that some matches do not work unless there are multiple question marks. 

### Code sample

```dart
String matchesTerm = '呼び?す';

termMatchHeadings = database.dictionaryHeadings
    .where()
    .filter()
    .termMatches(matchesTerm, caseSensitive: false)
    .limit(limit)
    .findAllSync();

/// No matches found for an item like  '呼び出す';
```

```dart
String matchesTerm = '呼び???す';

termMatchHeadings = database.dictionaryHeadings
    .where()
    .filter()
    .termMatches(matchesTerm, caseSensitive: false)
    .limit(limit)
    .findAllSync();

/// Match found for an item like  '呼び出す';
```
### Details

 - Platform: Android 13, Pixel 6 Pro
 - Flutter version: 3.7
 - Isar version: 3.0.6

---

 - [X] I searched for similar issues already
 - [X] I filled the details section with the exact device model and version
 - [X] I am able to provide a reproducible example

**Comment by CaptainDario** [comment]
Encountered the same today, maybe it will be fixed in v4.0

**Comment by simc** [comment]
Unfortunately this is a limitation with the wildcard search engine. Considering unicode characters would be a big performance hit and increase binary size. SQLite has the same behavior.
